### PR TITLE
Clarify unsafeTeleportDestination message

### LINKED ISSUE #5088: Incorrect teleportation safety message
### Type of bug

Other unexpected behaviour

### `/ess dump all` output

https://essentialsx.net/dump.html?id=c9ffe41de9fc4828b77c65c519d1b505

### Error log (if applicable)

_No response_

### Bug description

When you try to set a home in an "unsafe" location and you are in survival mode, it says this:

image

However, teleport-safety in my config is set to true. 

### Steps to reproduce

1. Stand half off a block (so the location is considered unsafe) and try to set a home
2. It will say it is unsafe and teleport-safety is disabled despite it not being disabled

### Expected behaviour

It should set the home

### Actual behaviour

It doesn't set the home

### PR DESCRIPTION
**Pull request by Haarolean**
### Information

This PR fixes #5088. 

### Details

**Proposed fix:**    

1. I've analyzed the usage of `unsafeTeleportDestination` message in all the places, two of them seem to use the message correctly (`com/earth2me/essentials/Teleport.java:156` and `com/earth2me/essentials/AsyncTeleport.java:207`)
2. The problematic one in `com/earth2me/essentials/commands/Commandsethome.java:59` needed some clarification as the message could be triggered via `isForceDisableTeleportSafety` being disabled, for example. I've altered the wording of the message to prevent adding additional messages or passing config param states into i18n.
3. Refactored the if clause in `Commandsethome.java` to make it more readable.

**Environments tested:**    

OS: 

Java version:  

- [ ] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [ ] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

**Demonstration:**

### CODE REVIEW

File: Essentials/src/main/resources/messages_ru.properties [hunk 1]
```diff
@@ -1452,7 +1452,7 @@ unlinkCommandUsage=/<command>
 unlinkCommandUsage1=/<command>
 unlinkCommandUsage1Description=Отвязывает ваш аккаунт Minecraft от вашего аккаунта Discord.
 unmutedPlayer=§6Игрок§c {0} §6снова может писать в чат.
-unsafeTeleportDestination=§4Место назначения телепортации небезопасно, а защита при телепортации отключена.
+unsafeTeleportDestination=§4Место назначения телепортации небезопасно, и правила защиты при телепортации не позволяют обойти ограничение.
```
Comment (mdcfe) [inline Essentials/src/main/resources/messages_ru.properties]: Only edit `messages.properties` in PRs, not translations - these will get overwritten by Crowdin

File: Essentials/src/main/resources/messages_en_GB.properties [hunk 2]
```diff
@@ -1432,7 +1432,7 @@ unlinkCommandUsage=/<command>
 unlinkCommandUsage1=/<command>
 unlinkCommandUsage1Description=Unlinks your Minecraft account from the currently linked Discord account.
 unmutedPlayer=§6Player§c {0} §6unmuted.
-unsafeTeleportDestination=§4The teleport destination is unsafe and teleport-safety is disabled.
+unsafeTeleportDestination=§4The teleport destination is unsafe and teleport-safety rules are preventing you from bypassing this.
```
Comment (mdcfe) [inline Essentials/src/main/resources/messages_en_GB.properties]: Same here - don't edit this file

File: Essentials/src/main/resources/messages.properties [hunk 3]
```diff
@@ -1453,7 +1453,7 @@ unlinkCommandUsage=/<command>
 unlinkCommandUsage1=/<command>
 unlinkCommandUsage1Description=Unlinks your Minecraft account from the currently linked Discord account.
 unmutedPlayer=\u00a76Player\u00a7c {0} \u00a76unmuted.
-unsafeTeleportDestination=\u00a74The teleport destination is unsafe and teleport-safety is disabled.
+unsafeTeleportDestination=\u00a74The teleport destination is unsafe and teleport-safety rules are preventing you from bypassing this.
```
Comment (mdcfe) [inline Essentials/src/main/resources/messages.properties:1456]: I agree `teleport-safety is disabled` is poorly worded, but this message also feels clunky

File: Essentials/src/main/resources/messages_ru.properties (hunk 1 above)
Comment (Haarolean) [reply Essentials/src/main/resources/messages_ru.properties]: Thanks, reverted

File: Essentials/src/main/resources/messages_en_GB.properties (hunk 2 above)
Comment (Haarolean) [reply Essentials/src/main/resources/messages_en_GB.properties]: Reverted

File: Essentials/src/main/resources/messages.properties (hunk 3 above)
Comment (Haarolean) [reply Essentials/src/main/resources/messages.properties:1456]: What about "The teleport destination is unsafe and unable to bypass teleport-safety rules"?

**Comment by Haarolean** [comment]
Welp thanks for the review I guess

**Comment by mdcfe** [comment]
Apologies for the slow response on this issue - I've had limited time to work on EssentialsX recently and this fell between the cracks during MC updates. I'm still open to improving where this message shows if you were interested in reopening this PR.

In the context of `/sethome` it does feel weird to use the same message as when we are trying to trigger a teleport - `/sethome` should probably have its own string worded such that "you can't set a home here as the teleport location would be unsafe" (and if `/setwarp` perform this check, similar message there).

Another note is that the if statement refactor means an expensive teleport safety check (scanning potentially many block locations) will be performed before checking the config - it would be ideal if this could be avoided.

**Pull request closed without merging**